homeassistant/components/ipma/sensor.py

```diff
@@ -89,11 +105,29 @@ def __init__(
         IPMADevice.__init__(self, api, location)
         self.entity_description = description
         self._attr_unique_id = f"{self._location.station_latitude}, {self._location.station_longitude}, {self.entity_description.key}"
+        self._ipma_data: Any | None = None
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     async def async_update(self) -> None:
         """Update sensors."""
         async with asyncio.timeout(10):
-            self._attr_native_value = await self.entity_description.value_fn(
+            self._ipma_data = await self.entity_description.value_fn(
                 self._location, self._api
             )
+
+    @property
+    def native_value(self) -> str | None:
+        """Return the state of the sensor."""
+        if self.entity_description.value_extractor is not None:
+            return self.entity_description.value_extractor(self._ipma_data)
+        return self._ipma_data
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes."""
+        if (
+            self.entity_description.value_extractor is not None
```

Comment:
It does indeed, but the issue is the API provides all the information (state and extra-attributes) in a single request and I must split that information into state and extra-attributes.

Reply:
I understood that part, that's why I suggested the second version.  
Add another function in IPMASensorEntityDescription to extract the attributes and use it instead of using something which might not be related

```
class IPMASensorEntityDescription(SensorEntityDescription):
    """Describes a IPMA sensor entity."""

    ....
    extra_attr_fn: Callable[[Any], dict[str, Any]] | None = None
```

Reply:
addressed